lab_3/main.py

```diff
@@ -31,39 +53,60 @@ class LetterStorage:
 
     def __init__(self):
         self.storage = {}
+        self.count = 1
 
     def _put_letter(self, letter: str) -> int:
         """
         Puts a letter into storage, assigns a unique id
         :param letter: a letter
         :return: 0 if succeeds, 1 if not
         """
-        pass
+        if not isinstance(letter, str):
+            return -1
+        if letter not in self.storage:
+            self.storage[letter] = self.count
+            self.count += 1
+        return 0
 
     def get_id_by_letter(self, letter: str) -> int:
         """
         Gets a unique id by a letter
         :param letter: a letter
         :return: an id
         """
-        pass
+        if not isinstance(letter, str):
+            return -1
+        return self.storage[letter]
 
     def get_letter_by_id(self, letter_id: int) ->str or int:
         """
         Gets a letter by a unique id
         :param letter_id: a unique id
         :return: letter
         """
-        pass
+        if not isinstance(letter_id, int):
+            return -1
+        for letter, new_letter_id in self.storage.items():
+            if new_letter_id == letter_id:
+                return letter
 
     def update(self, corpus: tuple) -> int:
         """
         Fills a storage by letters from the corpus
         :param corpus: a tuple of sentences
         :return: 0 if succeeds, 1 if not
         """
-        pass
-
+        if not isinstance(corpus, tuple):
+            return -1
+        for sentence in corpus:
+            for word in sentence:
+                for letter in word:
+                    if self._put_letter(letter) != -1:
```

Comment:
normally, we do not write code like this. When you mention method in if statement - you literally call it and get the result - can you rewrite this statement without calling the _put_letter method?

Reply:
found this